src/HotChocolate/ElasticSearch/src/Data.ElasticSearch.Driver/Filters/Handlers/ElasticSearchStringContainsOperationHandler.cs

```diff
@@ -0,0 +1,54 @@
+using HotChocolate.Configuration;
+using HotChocolate.Data.Filters;
+using HotChocolate.Language;
+using HotChocolate.Types;
+
+namespace HotChocolate.Data.ElasticSearch.Filters;
+
+/// <summary>
+/// This filter operation handler maps a Contains operation field to a
+/// <see cref="ISearchOperation"/>
+/// </summary>
+public class ElasticSearchStringContainsOperationHandler
+    : ElasticSearchOperationHandlerBase
+{
+    /// <summary>
+    /// Initializes a new instance of
+    /// <see cref="ElasticSearchStringContainsOperationHandler"/>
+    /// </summary>
+    public ElasticSearchStringContainsOperationHandler(InputParser inputParser)
+        : base(inputParser)
+    {
+    }
+
+    /// <inheritdoc />
+    public override bool CanHandle(
+        ITypeCompletionContext context,
+        IFilterInputTypeDefinition typeDefinition,
+        IFilterFieldDefinition fieldDefinition)
+        => context.Type is StringOperationFilterInputType &&
+            fieldDefinition is FilterOperationFieldDefinition
+            {
+                Id: DefaultFilterOperations.Contains
+            };
+
+    /// <inheritdoc />
+    public override ISearchOperation HandleOperation(
+        ElasticSearchFilterVisitorContext context,
+        IFilterOperationField field,
+        IValueNode value,
+        object? parsedValue)
+    {
+        if (parsedValue is not string val)
+        {
+            throw ThrowHelper.Filtering_WrongValueProvided(field);
+        }
+
+        IElasticFilterMetadata metadata = field.GetElasticMetadata();
+
+        return new ContainsOperation(
```

Comment:
Can we do 
```
Value = $"*{operation.Value}*"
``` 
here isntead of the rewriter?
If we do it in the rewriter we have to do it twice, once of elastic and once for opernsearch. lets just do it here once. 
Also, We have to escape the value. So when someone provides `where: {name: {contains: "foo*bar"}}` then we have to rewrite the value to "foo\*bar".
Possible characters to escape `+ - = && || > < ! ( ) { } [ ] ^ " ~ * ? : \ /` 
we need to check what we need to escape for the `Wildcard` operation :)

Reply:
Good point! Thanks for noticing ;)

I need a bit more explanations though. First of all, according to [elastic query documents](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html), the query supports 2 wildcard operations: `*` and `?`. Therefore I think we only need to escape these two, right?

Secondly, if someone provides `where: {name: {contains: "foo*bar"}}`, to my understanding, we need to escape the `*` in the middle of the string so that it gets treated as the actual asterisk character and still add the two `*` in the beginning and the end of the string. To what extend is this algorithm true?

Reply:
yes that seems right. But keep in mind, we also have to escape the escpae character `\`